.github/workflows/pr.yaml

```
@@ -50,6 +53,31 @@ jobs:
+          path: build/web
+          retention-days: 1
+
+  deploy-web:
+    # The only job that waits for the tests, because publishing a preview is
+    # the only step whose cost is not paid purely in CI minutes.
+    needs: [pre-build, build-web]
+    runs-on: ubuntu-latest
+    steps:
+      # `firebase.json` decides what is served and with which headers, so the
+      # deploy needs the repo as well as the build output.
+      - uses: actions/checkout@v3
```

actionlint (1.7.12)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)